setup.py

```diff
@@ -19,83 +19,13 @@
 """Setup for zope.i18nmessageid package
 """
 
-import os
 import sys
-
-from setuptools import setup, find_packages, Extension, Feature
-from distutils.command.build_ext import build_ext
-from distutils.errors import CCompilerError
-from distutils.errors import DistutilsExecError
-from distutils.errors import DistutilsPlatformError
-import platform
-
-py_impl = getattr(platform, 'python_implementation', lambda: None)
-is_pypy = py_impl() == 'PyPy'
-is_jython = 'java' in sys.platform
-
-codeoptimization_c = os.path.join('src', 'zope', 'i18nmessageid',
-                                  "_zope_i18nmessageid_message.c")
-codeoptimization = Feature(
-    "Optional code optimizations",
-    standard=True,
-    ext_modules=[Extension(
-        "zope.i18nmessageid._zope_i18nmessageid_message",
-        [os.path.normcase(codeoptimization_c)]
-        )])
-
-extra = {
-    'extras_require': {
-        'testing': ['nose', 'coverage'],
-        'docs': ['Sphinx'],
-    },
-}
-
-if not is_pypy and not is_jython:
-    # Jython cannot build the C optimizations, while on PyPy they are
-    # anti-optimizations (the C extension compatibility layer is known-slow,
-    # and defeats JIT opportunities).
-    extra['features'] = {'codeoptimization': codeoptimization}
+from os import path
+from setuptools import setup, find_packages
 
 
 def read(*rnames):
-    return open(os.path.join(os.path.dirname(__file__), *rnames)).read()
-
-
-class optional_build_ext(build_ext):
-    """This class subclasses build_ext and allows
-       the building of C extensions to fail.
-    """
-    def run(self):
-        try:
-            build_ext.run(self)
-
-        except DistutilsPlatformError:
-            # The sys.exc_info()[1] is to preserve compatibility with both
-            # Python 2.5 and 3.x, which is needed in setup.py.
-            self._unavailable(sys.exc_info()[1])
-
-    def build_extension(self, ext):
-        try:
-            build_ext.build_extension(self, ext)
-
-        except (CCompilerError, DistutilsExecError):
-            # The sys.exc_info()[1] is to preserve compatibility with both
-            # Python 2.5 and 3.x, which is needed in setup.py.
-            self._unavailable(sys.exc_info()[1])
-
-    def _unavailable(self, e):
-        # Write directly to stderr to preserve compatibility with both
-        # Python 2.5 and 3.x, which is needed in setup.py.
-        sys.stderr.write('*' * 80 + '\n')
-        sys.stderr.write("""WARNING:
-
-        An optional code optimization (C extension) could not be compiled.
-
-        Optimizations for this package will not be available!
-
-        """)
-        sys.stderr.write(str(e) + '\n')
-        sys.stderr.write('*' * 80 + '\n')
+    return open(path.join(path.dirname(__file__), *rnames)).read()
```

Comment:
What a perfect opportunity to use a `with open(...) as f: return f.read()` here!